From 52e45023034c81531632be9304921a0bda576a2a Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob2.chen@rock-chips.com>
Date: Thu, 23 Nov 2017 14:18:41 +0800
Subject: [PATCH] media: i2c: tc35874x: add link freq control

before we get it from dts, but now it must be get from driver

Change-Id: I911f02a1c4d6618c539aceb06d485b58eb4906cd
Signed-off-by: Jacob Chen <jacob2.chen@rock-chips.com>
---
 drivers/media/i2c/tc35874x.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/media/i2c/tc35874x.c b/drivers/media/i2c/tc35874x.c
index d6a063e7273f..283fe8dbbc7d 100644
--- a/drivers/media/i2c/tc35874x.c
+++ b/drivers/media/i2c/tc35874x.c
@@ -66,6 +66,10 @@ MODULE_LICENSE("GPL");
 
 #define POLL_INTERVAL_MS	1000
 
+static const s64 link_freq_menu_items[] = {
+	300000000,
+};
+
 static const struct v4l2_dv_timings_cap tc35874x_timings_cap = {
 	.type = V4L2_DV_BT_656_1120,
 	/* keep this initialization for compatibility with GCC < 4.4.6 */
@@ -1926,6 +1930,9 @@ static int tc35874x_probe(struct i2c_client *client,
 	state->audio_present_ctrl = v4l2_ctrl_new_custom(&state->hdl,
 			&tc35874x_ctrl_audio_present, NULL);
 
+	v4l2_ctrl_new_int_menu(&state->hdl, NULL, V4L2_CID_LINK_FREQ,
+			       0, 0, link_freq_menu_items);
+
 	sd->ctrl_handler = &state->hdl;
 	if (state->hdl.error) {
 		err = state->hdl.error;
-- 
2.35.3

